MapleServer2/Data/Static/MapMetadataStorage.cs

```diff
@@ -73,8 +89,52 @@ public static int GetPlotNumber(int mapId, CoordB coord)
             {
                 return block.SaleableGroup;
             }
+
             coordS.Z -= Block.BLOCK_SIZE;
         }
+
         return 0;
     }
+
+    public static bool IsLiquidBlock(int mapId, CoordS coord)
+    {
+        MapBlock block = GetMapBlock(mapId, coord);
+        return block is not null && IsLiquidBlock(block);
+    }
+
+    public static bool IsLiquidBlock(MapBlock block)
+    {
+        if (block.Type == "Ground")
+        {
+            return false;
+        }
+
+        return block.Attribute is "water" or "seawater" or "devilwater" or "lava" or "poison" or "oil" or "emeraldwater";
+    }
+
+    public static int GetDistanceToNextBlockBelow(int mapId, CoordS coord, out MapBlock block)
+    {
+        CoordS tempCoord = coord;
+        MapMetadata mapMetadata = GetMetadata(mapId);
+        if (mapMetadata is null)
+        {
+            block = null;
+            return 99;
+        }
+
+        // checking 20 blocks in the same Z axis
+        for (int i = 0; i < 20; i++)
```

Comment:
is there really a need to check 20 blocks considering the skills only go down up to 2?